### DISCUSSION TITLE
Allow arbitrary objects in app.storage, but only persist serializable data

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#2841] [Ideas / Feature Requests]
The idea came up in https://github.com/zauberzeug/nicegui/pull/2837#issuecomment-2041162862: `app.storage.user` and `app.storage.general` currently only work with serializable data. The pull request #2837 introduces `app.storage.tab` and does not do persistence (yet) because it is really handy to save arbitrary data types like images, plot-data etc on a per-tab basis (similar to `app.storage.client` from #2820.

To combine persistence across server restarts and the flexibility of arbitrary data types we could modify our PersistentDict to only saves all serializable fields to disc and simply skip everything else.

**Comment by falkoschindler** [comment]
This sounds like a rather unpredictable behavior for the unexperienced user, who doesn't necessarily know what data can be serialized and what can't. I don't really have a better idea though - apart from maybe explicitly providing a way to configure which data to persist. 🤔
